src/dodal/common/udc_directory_provider.py

```diff
@@ -20,7 +24,13 @@ def __init__(self, directory: Path | None = None):
             else None
         )
 
-    def update(self, directory: Path):
+    def update(self, directory: Path, create_directory=False):
+        if create_directory:
```

Comment:
As is, you're allowed to (but always not) pass in a Path on construction, but the path is only created on update. If passing a Path into the initialiser isn't desired, remove it, if it is desired sometimes, make sure that the directory may be created during the init